internal/errors/errors.go

```diff
@@ -26,48 +26,71 @@ import (
 )
 
 var (
+	// ErrTimeoutParseFailed represents a function to generate an error that the timeout value is invalid.
 	ErrTimeoutParseFailed = func(timeout string) error {
 		return Errorf("invalid timeout value: %s\t:timeout parse error out put failed", timeout)
 	}
 
+	// ErrServerNotFound represents a function to generate an error that the server not found.
 	ErrServerNotFound = func(name string) error {
 		return Errorf("server %s not found", name)
 	}
 
+	// ErrOptionFailed represents a function to generate an error that the option setup failed.
+	// When ref is zero value, it will return an error with ref is invalid.
 	ErrOptionFailed = func(err error, ref reflect.Value) error {
-		return Wrapf(err, "failed to setup option :\t%s",
-			runtime.FuncForPC(ref.Pointer()).Name())
+		var str string
+		if ref.IsValid() {
+			str = runtime.FuncForPC(ref.Pointer()).Name()
+		}
+		return Wrapf(err, "failed to setup option :\t%s", str)
 	}
 
+	// ErrArgumentParseFailed represents a function to generate an error that argument parse failed.
 	ErrArgumentParseFailed = func(err error) error {
 		return Wrap(err, "argument parse failed")
 	}
 
+	// ErrBackoffTimeout represents a function to generate an error that backoff is timeout by limitation.
 	ErrBackoffTimeout = func(err error) error {
 		return Wrap(err, "backoff timeout by limitation")
 	}
 
+	// ErrInvalidTypeConversion represents a function to generate an error that type conversion fails due to an invalid input type.
 	ErrInvalidTypeConversion = func(i interface{}, tgt interface{}) error {
 		return Errorf("invalid type conversion %v to %v", reflect.TypeOf(i), reflect.TypeOf(tgt))
 	}
 
+	// ErrLoggingRetry represents a function to generate an error that failing to output logs and retrying to output.
 	ErrLoggingRetry = func(err error, ref reflect.Value) error {
-		return Wrapf(err, "failed to output %s logs, retrying...",
-			runtime.FuncForPC(ref.Pointer()).Name())
+		var str string
+		if ref.IsValid() {
+			str = runtime.FuncForPC(ref.Pointer()).Name()
+		}
+		return Wrapf(err, "failed to output %s logs, retrying...", str)
 	}
 
+	// ErrLoggingFailed represents a function to generate an error that failing to output logs.
 	ErrLoggingFailed = func(err error, ref reflect.Value) error {
-		return Wrapf(err, "failed to output %s logs",
-			runtime.FuncForPC(ref.Pointer()).Name())
+		var str string
+		if ref.IsValid() {
+			str = runtime.FuncForPC(ref.Pointer()).Name()
+		}
+		return Wrapf(err, "failed to output %s logs", str)
 	}
 
+	// New represents a function to generate the new error with a message.
+	// When the message is nil, it will return nil instead of an error.
 	New = func(msg string) error {
 		if msg == "" {
 			return nil
 		}
 		return errors.New(msg)
 	}
 
+	// Wrap represents a function to generate an error that is used by input error and message.
+	// When both of the input is nil, it will return a new error with the message even message is nil.
```

Comment:
I think it is incorrect, the nil will be returned if the message is nil (line 85)
```suggestion
	// When both of the input are nil, it will return an error when the error message is not empty.
```